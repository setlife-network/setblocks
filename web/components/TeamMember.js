import React from 'react';
import { Edit3 } from 'styled-icons/feather/Edit3.cjs'

import Card from './Card';
import Flex from './Flex';
import Text from './Text';

import theme from '../styles/theme';


export default class TeamMember extends React.Component {
    render() {
        const { goToPage, name, color, goToEdit } = this.props;
        
        return (
            <>
                <Card
                    bg='white'
                    width='90%'
                    minWidth='20%'
                    maxWidth='530px'
                    height='60px'
                    mx='auto'
                    mt='1.5rem'
                    depth={1}
                >
                    <Flex row center px='1rem'>
                        <Flex
                            direction='row'
                            mb='2rem'
                            width='100%'
                            height='60px'
                            onClick={goToPage}
                        >
                            <Card
                                width='10px'
                                height='10px'
                                bg={color}
                                borderRadius={50}
                                my='auto'
                                mx='5%'
                            />
                            <Text
                                color='textPrimary'
                                size='1rem'
                                my='auto'
                                weight='600'
                                ml='5%'
                            >
                                {name}
                            </Text>
                        </Flex>
                        <Flex height='60px'>
                            { // Workaround for edit until to have a login
                                React.createElement(Edit3, {
                                    size: 24,
                                    color: theme.colors.charcoal,
                                    onClick: goToEdit
                                })
                            }
                        </Flex>
                    </Flex>
                </Card>
            </>
        );
    }
}
