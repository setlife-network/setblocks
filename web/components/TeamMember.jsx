import React from 'react';
import { Edit3 } from 'styled-icons/feather/Edit3.cjs'

import Card from './Card';
import Flex from './Flex';
import Text from './Text';

import theme from '../styles/theme';


export default class TeamMember extends React.Component {
    render() {
        const { goToPage, name, color, goToEdit } = this.props;
        return (
            <>
                <Card
                    bg='primary'
                    width='95%'
                    minWidth='20%'
                    maxWidth='530px'
                    height='60px'
                    mx='auto'
                    mt='1rem'
                    depth={7}
                >
                    <Flex row center>
                        <Flex
                            direction='row'
                            mb='2rem'
                            width='100%'
                            height='60px'
                            onClick={goToPage}
                        >
                            <Card
                                width='10px'
                                height='10px'
                                bg={color}
                                borderRadius={50}
                                my='auto'
                                ml='20%'
                            />
                            <Text
                                color='textPrimary'
                                size='1rem'
                                weight='600'
                                ml='15%'
                            >
                                {name}
                            </Text>
                        </Flex>
                        <Flex height='60px' mr='50px'>
                            { // Workaround for edit until to have a login
                                React.createElement(Edit3, {
                                    size: 24,
                                    color: theme.colors.lightBlue,
                                    onClick: goToEdit
                                })
                            }
                        </Flex>
                    </Flex>
                </Card>
            </>
        );
    }
}
