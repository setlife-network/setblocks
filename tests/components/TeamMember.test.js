import React from 'react';
import { shallow } from 'enzyme';
import TeamMember from '../../web/components/TeamMember';

import Text from '../../web/components/Text';

import { MOCK_TEAM_MEMBER } from '../data'

let renderedComponent;

let mock = () => jest.fn();

describe('TeamMember Component', () => {
    beforeEach(() => {

        renderedComponent = shallow(
            <TeamMember
                name={MOCK_TEAM_MEMBER.name}
            />
        );
    });

    it('should render correctly', () => {
        expect(renderedComponent).toMatchSnapshot();
    })

    it('should display the correct name', () => {
        let nameText = renderedComponent.find(Text)
        
        expect(nameText.contains(MOCK_TEAM_MEMBER.name)).toBe(true)
    })

});