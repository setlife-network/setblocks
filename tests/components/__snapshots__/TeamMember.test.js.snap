// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamMember Component should render correctly 1`] = `
<Fragment>
  <Styled(styled.div)
    bg="primary"
    depth={7}
    height="60px"
    maxWidth="530px"
    minWidth="20%"
    mt="1rem"
    mx="auto"
    width="95%"
  >
    <Flex
      center={true}
      row={true}
    >
      <Flex
        direction="row"
        height="60px"
        mb="2rem"
        width="100%"
      >
        <Styled(styled.div)
          borderRadius={50}
          height="10px"
          ml="20%"
          my="auto"
          width="10px"
        />
        <Text
          color="textPrimary"
          fontFamily="Nunito,sans-serif"
          letterSpacing="0.025rem"
          lineHeight={1.6}
          ml="15%"
          size="1rem"
          weight="600"
        >
          Team Member #1
        </Text>
      </Flex>
      <Flex
        height="60px"
        mr="50px"
      >
        <Edit3
          color="#8CC3E6"
          size={24}
        />
      </Flex>
    </Flex>
  </Styled(styled.div)>
</Fragment>
`;
